src/Core/LanguageService.fs

```diff
@@ -650,13 +668,73 @@ Consider:
-        let spawnNetCore dotnet =
+
+
+
+        let discoverDotnetArgs () = promise {
+            let! (rollForwardArgs, necessaryEnvVariables) = promise {
+                let! sdkVersionAtRootPath = runtimeVersion()
+                match sdkVersionAtRootPath with
+                | Error e ->
+                    printfn $"FSAC (NETCORE): {e}"
+                    return [], []
+                | Ok v ->
+                    if v.major >= 6.0
+                    then
+                        // when we run on a sdk higher than 5.x (aka what FSAC is currently built/targeted for),
+                        // we have to tell the runtime to allow it to actually run on that runtime (instead of presenting 6.x as 5.x)
+                        // in order for msbuild resolution to work
+                        let args = ["--roll-forward"; "LatestMajor"]
+                        let envs =
+                            if v.prerelease <> null || v.prerelease.Count > 0
+                            then ["DOTNET_ROLL_FORWARD_TO_PRERELEASE", box 1]
+                            else []
+                        return args, envs
+                    else return [], []
+            }
+            let userDotnetArgs = "FSharp.fsac.dotnetArgs" |> Configuration.get [||]
+            let hasUserRollForward = userDotnetArgs |> Array.tryFindIndex (fun a -> a = "--roll-forward") |> Option.map (fun _ -> true) |> Option.defaultValue false
+            let hasUserFxVersion = userDotnetArgs |> Array.tryFindIndex (fun a -> a = "--fx-version") |> Option.map (fun _ -> true) |> Option.defaultValue false
+            let shouldApplyImplicitRollForward = not (hasUserFxVersion || hasUserRollForward)
+
+            let args =
+                [
+                    if shouldApplyImplicitRollForward then yield! rollForwardArgs
+                    yield! userDotnetArgs
+                ]
+            let envVariables =
+                [
+                    if shouldApplyImplicitRollForward then yield! necessaryEnvVariables
+                ]
+
+            return args, envVariables
+        }
+
+        let spawnNetCore dotnet: JS.Promise<Executable> = promise {
             let fsautocompletePath =
                 if String.IsNullOrEmpty fsacNetcorePath
                 then VSCodeExtension.ionidePluginPath () + "/bin/fsautocomplete.dll"
                 else fsacNetcorePath
-            printfn "FSAC (NETCORE): '%s'" fsautocompletePath
+
+            let! (fsacDotnetArgs, fsacEnvVars) = discoverDotnetArgs()
+
+            printfn $"FSAC (NETCORE): '%s{fsautocompletePath}'"
+
+            let exeOpts = createEmpty<ExecutableOptions>
+            let exeEnv =
+                match fsacEnvVars with
+                | [] -> None
+                | fsacEnvVars ->
+                    // only need to set the process env if FSAC needs rollfoward env vars.
+                    let keys = Node.Util.Object.keys process.env
```

Comment:
we seemingly have to copy over the current process environment, even though my reading of the source code suggests otherwise.  if I didn't do this then cracking failed every time.